lte/gateway/c/core/oai/tasks/nas/nas_proc.c

```diff
@@ -578,13 +583,15 @@ status_code_e nas_proc_downlink_unitdata(
 
   IMSI_STRING_TO_IMSI64(dl_unitdata->imsi, &imsi64);
 
-  OAILOG_DEBUG(LOG_NAS_EMM, "Handling imsi " IMSI_64_FMT "\n", imsi64);
+  OAILOG_DEBUG_UE(
+      LOG_NAS_EMM, ctxt->_imsi64, "Handling imsi " IMSI_64_FMT "\n", imsi64);
 
   ctxt = emm_context_get_by_imsi(&_emm_data, imsi64);
 
   if (!(ctxt)) {
-    OAILOG_ERROR(
-        LOG_NAS_EMM, "That's embarrassing as we don't know this IMSI\n");
+    OAILOG_ERROR_UE(
+        LOG_NAS_EMM, ctxt->_imsi64,
```

Comment:
you cannot use ctxt, you can replace ctxt->_imsi64 with imsi64

Reply:
@Vyas20 if imsi64 is available let's use that instead of accessing the ctxt as Lionel states

Reply:
Resolved